pkgs/development/python-modules/web3/default.nix

```diff
@@ -33,24 +33,30 @@ buildPythonPackage rec {
     hash = "sha256-RNWCZQjcse415SSNkHhMWckDcBJGFZnjisckF7gbYY8=";
   };
 
+  # Note: to reflect the extra_requires in main/setup.py.
+  passthru.optional-dependencies = {
+    # tester = [ eth-tester py-geth ];
+    # linter = [ ... ];
+    # docs = [ ... ];
+    # dev = [ ... ];
+    ipfs = [ ipfshttpclient ];
+  };
+
   propagatedBuildInputs = [
     aiohttp
     eth-abi
     eth-account
-    eth-hash
-    eth-rlp
+    eth-hash ] ++ eth-hash.optional-dependencies.pycryptodome ++ [
```

Comment:
I have never seen this writing style and don't feel any good reason to change it.